python/manylinux1/build_arrow.sh

```diff
@@ -48,95 +45,73 @@ export PYARROW_CMAKE_OPTIONS='-DTHRIFT_HOME=/usr -DBoost_NAMESPACE=arrow_boost -
-
-    echo "=== (${PYTHON}) Install modules required for testing ==="
-    pip install -r requirements-test.txt
-
-    # The TensorFlow test will be skipped here, since TensorFlow is not
-    # manylinux1 compatible; however, the wheels will support TensorFlow on
-    # a TensorFlow compatible system
-    py.test -v -r sxX --durations=15 --parquet ${VIRTUAL_ENV}/lib/*/site-packages/pyarrow
-    deactivate
-
-    mv repaired_wheels/*.whl /io/dist
-    mv dist/*.tar.gz /io/dist
-done
+# Must pass PYTHON_VERSION and UNICODE_WIDTH env variables
+# possible values are: 2.7,16 2.7,32 3.5,16 3.6,16 3.7,16
+
+CPYTHON_PATH="$(cpython_path ${PYTHON_VERSION} ${UNICODE_WIDTH})"
+PYTHON_INTERPRETER="${CPYTHON_PATH}/bin/python"
+PIP="${CPYTHON_PATH}/bin/pip"
+PATH="${PATH}:${CPYTHON_PATH}"
+
+if [ ${PYTHON_VERSION} != "2.7" ]; then
+  # Gandiva is not supported on Python 2.7
+  export PYARROW_WITH_GANDIVA=1
+  export BUILD_ARROW_GANDIVA=ON
+else
+  export PYARROW_WITH_GANDIVA=0
+  export BUILD_ARROW_GANDIVA=OFF
+fi
+
+echo "=== (${PYTHON_VERSION}) Building Arrow C++ libraries ==="
+ARROW_BUILD_DIR=/tmp/build-PY${PYTHON_VERSION}-${UNICODE_WIDTH}
+mkdir -p "${ARROW_BUILD_DIR}"
+pushd "${ARROW_BUILD_DIR}"
+PATH="${CPYTHON_PATH}/bin:${PATH}" cmake -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=/arrow-dist \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DARROW_BUILD_TESTS=OFF \
+    -DARROW_BUILD_SHARED=ON \
+    -DARROW_BOOST_USE_SHARED=ON \
+    -DARROW_GANDIVA_PC_CXX_FLAGS="-isystem;/opt/rh/devtoolset-2/root/usr/include/c++/4.8.2;-isystem;/opt/rh/devtoolset-2/root/usr/include/c++/4.8.2/x86_64-CentOS-linux/" \
+    -DARROW_JEMALLOC=ON \
+    -DARROW_RPATH_ORIGIN=ON \
+    -DARROW_PYTHON=ON \
+    -DARROW_PARQUET=ON \
+    -DPythonInterp_FIND_VERSION=${PYTHON_VERSION} \
+    -DARROW_PLASMA=ON \
+    -DARROW_TENSORFLOW=ON \
+    -DARROW_ORC=ON \
+    -DARROW_GANDIVA=${BUILD_ARROW_GANDIVA} \
+    -DARROW_GANDIVA_JAVA=OFF \
+    -DBoost_NAMESPACE=arrow_boost \
+    -DBOOST_ROOT=/arrow_boost_dist \
+    -GNinja /arrow/cpp
+ninja install
+popd
+
+# Check that we don't expose any unwanted symbols
+/io/scripts/check_arrow_visibility.sh
+
+echo "=== (${PYTHON_VERSION}) Install the wheel build dependencies ==="
+$PIP install -r requirements-wheel.txt
+
+# Clear output directory
+rm -rf dist/
+echo "=== (${PYTHON_VERSION}) Building wheel ==="
+# Remove build directory to ensure CMake gets a clean run
+rm -rf build/
+PATH="$PATH:${CPYTHON_PATH}/bin" $PYTHON_INTERPRETER setup.py build_ext \
+    --inplace \
+    --bundle-arrow-cpp \
+    --bundle-boost \
+    --boost-namespace=arrow_boost
+PATH="$PATH:${CPYTHON_PATH}/bin" $PYTHON_INTERPRETER setup.py bdist_wheel
+PATH="$PATH:${CPYTHON_PATH}/bin" $PYTHON_INTERPRETER setup.py sdist
+
+echo "=== (${PYTHON_VERSION}) Tag the wheel with manylinux1 ==="
+mkdir -p repaired_wheels/
+auditwheel -v repair -L . dist/pyarrow-*.whl -w repaired_wheels/
+
+# Testing happens outsite of the build to prevent issues like ARROW-4372
```

Comment:
Can you keep the import checks here? They are fast and already provide good insights.

Reply:
Good point!

Reply:
Added.